scaleway/loadbalancers.go

```diff
@@ -567,6 +572,46 @@ func (l *loadbalancers) unannotateAndPatch(service *v1.Service) error {
 }
 
 func (l *loadbalancers) updateLoadBalancer(ctx context.Context, loadbalancer *scwlb.LB, service *v1.Service, nodes []*v1.Node) error {
+	if l.pnID != "" {
+		respPN, err := l.api.ListLBPrivateNetworks(&scwlb.ZonedAPIListLBPrivateNetworksRequest{
+			Zone: loadbalancer.Zone,
+			LBID: loadbalancer.ID,
+		})
+		if err != nil {
+			return fmt.Errorf("error listing private networks of load balancer %s: %v", loadbalancer.ID, err)
+		}
+
+		var pnNIC *scwlb.PrivateNetwork
+		for _, pNIC := range respPN.PrivateNetwork {
+			if pNIC.PrivateNetworkID == l.pnID {
+				pnNIC = pNIC
+				break
```

Comment:
just a small detail, but breaking early here will not detach all unneeded private networks